src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -2330,11 +2331,11 @@ public B route(MessageProcessorSpec<?> messageProcessorSpec,
 					hasSubFlows = true;
 					IntegrationFlowDefinition<?> flowBuilder = (IntegrationFlowDefinition<?>) component;
 					addComponent(flowBuilder.fixedSubscriberChannel()
```

Comment:
M-m-m. Looks like otherwise: a bug in the `IntegrationFlowDefinition.registerOutputChannelIfCan()`:
```
else {
	throw new BeanCreationException("The 'currentComponent' (" + currentComponent +") is a one-way 'MessageHandler' and it isn't appropriate to configure 'outputChannel'. " +
			"This is the end of the integration flow.");
}
```
So, I think I can overcome it with `try...catch` :smile: 
... or just with simple `outputChannelRequired()` method...